core/src/OpenMPTarget/Kokkos_OpenMPTarget_Instance.cpp

```diff
@@ -164,8 +166,12 @@ void OpenMPTarget::impl_static_fence(const std::string& name) {
       name, Kokkos::Experimental::Impl::openmp_fence_is_static::yes);
 }
 
-void OpenMPTarget::impl_initialize(InitializationSettings const&) {
-  Impl::OpenMPTargetInternal::impl_singleton()->impl_initialize();
+void OpenMPTarget::impl_initialize(InitializationSettings const& settings) {
+  Impl::OpenMPTargetInternal::impl_singleton()->impl_initialize(settings);
+
+  using Kokkos::Impl::get_gpu;
+  const int device_num = get_gpu(settings);
+  omp_set_default_device(device_num);
```

Comment:
I would do it in that order to avoid the risk of bugs down the road
```suggestion
  using Kokkos::Impl::get_gpu;
  const int device_num = get_gpu(settings);
  omp_set_default_device(device_num);
  
    Impl::OpenMPTargetInternal::impl_singleton()->impl_initialize(settings);

```